src/components/pages/components/LinkResolver/Resolvers/StudyLogin.tsx

```diff
@@ -56,6 +64,12 @@ const StudyLogin: React.FC<StudyLoginProps> = (props) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  useEffect(() => {
+    if (loginInitiated && logedInUser.length > 0 && redirectTo !== undefined && redirectTo.length > 0) {
+      history.replace(redirectTo)
+    }
+  }, [logedInUser])
+
```

Comment:
Currently, the redirect logic is spread in at least two places (see also lines 87 - 91), we might want to centralized in one place so that it is easy to understand and maintain.

Also, this `useEffect` calls lacks some dependencies since it is also coupled with `loginInitiated` and `redirectTo`. This could be a source of problems (see: [exhaustive deps](https://github.com/facebook/react/issues/14920)).

Reply:
The logic should only be triggered when `loggedInUser`'s value changed, not when the others. In many cases this linter rule does not make sense, because it results in suboptimal code.

Reply:
I see what you mean, but [as stated by the developers](https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528) 

> Usually disabling it is a mistake and will significantly bite you later. (Most cases where people think it's safe to disable are not safe and lead to stale closures.)

You can see here, that if something changes in the way we update `redirectTo`, for example, might break this part of the code. 
See also [this list of cases where removing the dependencies created problems](https://github.com/facebook/react/issues/14920#issuecomment-471070149)

As for performance, I tend to usually prefer leaning towards the safe side and code maintainability unless there are serious performance implication worth considering.   

All in all, I think it's worth considering finding a good solution for the first point raised (centralizing the redirect logic), if we can come up with something good, it can improve the code a lot I think.

Reply:
Isn't this only relevant if we use some closures - e.g., we would like to define callbacks for setTimeout, or setState?

Between:
```
useEffect(() => {
    if (loginInitiated && loggedInUser.length > 0 && isRedirectToDefined()) {
      history.replace(redirectTo as string)
    }
  }, [loginInitiated, loggedInUser, history, redirectTo, isRedirectToDefined])

  useEffect(() => {
    if (loginInitiated && loggedInUser.length > 0 && isRedirectToDefined()) {
      history.replace(redirectTo as string)
    }
  }, [loggedInUser])
```

and 

```

  useEffect(() => {
    if (loginInitiated && loggedInUser.length > 0 && isRedirectToDefined()) {
      history.replace(redirectTo as string)
    }
  }, [loginInitiated, loggedInUser, history, redirectTo, isRedirectToDefined])
```

I cannot notice any difference in behaviour when testing it. But the former is much easier to understand, when it is intended to be used.

Reply:
No it's not about closure, but stale state and what happen if the code slightly changes breaking the assumptions that now seems to be valid. An unrelated change can break that functionality, and since the test is currently mostly done manually, this problem can go unnoticed up until it ends up in production. Right now there might not be a difference in behaviour as you have noticed